core/src/main/java/org/elasticsearch/search/aggregations/pipeline/movavg/models/EwmaModel.java

```diff
@@ -49,6 +50,28 @@ public EwmaModel(double alpha) {
         this.alpha = alpha;
     }
 
+    @Override
+    public <T extends Number> double[] predict(Collection<T> values, int numPredictions) {
+        double[] predictions = new double[numPredictions];
+
+        // If there are no values, we can't do anything.  Return an array of NaNs.
+        if (values.size() == 0) {
+            return emptyPredictions(numPredictions);
+        }
+
+        // special case for one prediction, avoids allocation
+        if (numPredictions < 1) {
+            throw new IllegalArgumentException("numPredictions may not be less than 1.");
+        } else if (numPredictions == 1){
```

Comment:
the `numPredications < 1` check feels like an assertion... should probably be the first statement in the method